server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -182,20 +192,23 @@ private void addUnmappedFields(String[] indices, String field, Map<String, Field
             FieldCapabilities.Builder unmapped = new FieldCapabilities.Builder(field, "unmapped");
             typeMap.put("unmapped", unmapped);
             for (String index : unmappedIndices) {
-                unmapped.add(index, false, false, Collections.emptyMap());
+                unmapped.add(index, false, false, false, Collections.emptyMap());
             }
         }
     }
 
     private void innerMerge(Map<String, Map<String, FieldCapabilities.Builder>> responseMapBuilder,
-                                String indexName, Map<String, IndexFieldCapabilities> map) {
-        for (Map.Entry<String, IndexFieldCapabilities> entry : map.entrySet()) {
+                            FieldCapabilitiesIndexResponse response) {
+        for (Map.Entry<String, IndexFieldCapabilities> entry : response.get().entrySet()) {
             final String field = entry.getKey();
+            // best effort to detect metadata field coming from older nodes
```

Comment:
you mention here that it's best effort: what are the cases where this may not be accurate? Would it be worth expanding the comment for future readers?

Reply:
See https://github.com/elastic/elasticsearch/pull/69977/commits/0b3c922b782ce57b14254601d65b98cde5c32f7c